cypress/e2e/audio-player/audio-player.spec.ts

```diff
@@ -0,0 +1,588 @@
+
+        // Wait until configuration is finished
+        cy.contains(
+            ".mx_RoomView_body .mx_GenericEventListSummary[data-layout='group'] .mx_GenericEventListSummary_summary",
+            "created and configured the room.",
+        ).should("exist");
+    };
+
+    const uploadFile = (file: string) => {
+        // Upload a file from the message composer
+        cy.get(".mx_MessageComposer_actions input[type='file']").selectFile(file, { force: true });
+
+        cy.get(".mx_Dialog").within(() => {
+            // Click "Upload" button
+            cy.get("[data-testid='dialog-primary-button']").should("have.text", "Upload").click();
+        });
+
+        // Wait until the file is sent
+        cy.get(".mx_RoomView_statusArea_expanded").should("not.exist");
+        cy.get(".mx_EventTile.mx_EventTile_last .mx_EventTile_receiptSent").should("exist");
+    };
+
+    // Check that the audio player for the long-name-audio-file has been rendered correctly
+    const checkPlayerFilenameLong = () => {
+        // Detect the audio file
+        cy.get(".mx_EventTile_mediaLine .mx_MAudioBody").within(() => {
+            // Assert that the audio player is rendered
+            cy.get(".mx_AudioPlayer_container").within(() => {
+                // Assert that media information is rendered
+                cy.get(".mx_AudioPlayer_mediaInfo").within(() => {
+                    cy.get(".mx_AudioPlayer_mediaName").should("have.text", "1sec-long-name-audio-file.ogg");
+                    cy.contains(".mx_AudioPlayer_byline", "00:01").should("exist");
+                    cy.contains(".mx_AudioPlayer_byline", "(3.56 KB)").should("exist"); // actual size
+                });
+
+                // Assert that the play button is rendered
+                cy.get("[data-testid='play-pause-button'][aria-label='Play']").should("exist");
+            });
+        });
+    };
+
+    // Check that the audio player is rendered and its button becomes visible
+    const checkPlayerVisibility = () => {
+        // Assert that the player is visible
+        cy.get(".mx_MAudioBody").should("be.visible");
+
+        // Assert that rendering of it settled and the play button is visible
+        cy.get(".mx_MAudioBody [data-testid='play-pause-button'][aria-label='Play']").should("be.visible");
+    };
+
+    /**
+     * Define snapshot widths for various scenarios.
+     *
+     * 267: 243px + 12px + 12px = 267px. Refer _MediaBody.pcss and _AudioPlayer.pcss
+     * 133: 267px / 2 = 133.5px
+     * 303: 243px + 12px + 48px = 303px. Refer _EventBubbleTile.pcss and _AudioPlayer.pcss
+     * 151: 303px / 2 = 151.5px
+     * 1024 and 1920: default values specified on .percy.yml
+     */
+    const snapshotWidthsGroup = [133, 267];
+    const snapshotWidthsIRC = snapshotWidthsGroup;
+    const snapshotWidthsBubble = [151, 303];
+    const snapshotWidthsEventTile = [133, 151, 267, 303, 1024, 1920];
+
```

Comment:
Those values are required for emulating the sizes of `mx_MAudioBody`, which `mx_EventTile` implements as the audio player.  `267px` is the actual size of `mx_MAudioBody` rendered on group/modern layout, and `303px` is on the bubble layout. You should be able to confirm those values by uploading an audio file to a room and checking it on the browser's devtool. 133px and 151px are (almost) the half size of them, respectively (`267 / 2 = 133.5`). Since they are magic numbers to check whether the player would be rendered properly even if it is compressed on a narrow timeline (please refer: https://github.com/vector-im/element-web/issues/19734), they could be other values. I thought 50% would be a simple and good start.

The important thing is that specifying custom values on `percySnapshotElement` has it Percy take snapshots of the *specified element in those widths*, not the default sizes specified on [these lines on `.percy.yml`](https://github.com/matrix-org/matrix-react-sdk/blob/d821323e5cb5f57231b91cdc7887c249ec522ec9/.percy.yml#L4-L5) ([this page](https://docs.percy.io/docs/cli-configuration#per-snapshot-configuration) looks to be the official documentation about it). This means that unless you specify a custom value, `percySnapshotElement` takes snapshots of the audio player in the default values. Obviously, we would want to avoid that, as the nobody sees "1024px / 1920px size audio player" on the actual UI. We would instead want to test the player's layout in the actual size rendered on the viewport, to check if a string, a button or something like this overflows it.

The values of `snapshotWidthsEventTile` is intended to have Percy take a snapshot of `mx_EventTile`, on which `mx_MAudioBody` is rendered, in those values. The smallest one is 133px (50% of the audio player on `mx_EventTile` on group/modern layout), and the largest one is 1920px. Plase note that those values are supposed to be used for capturing `mx_EventTile`, not the audio player itself rendered on `mx_EventTile`.

Here is an existing example: https://github.com/matrix-org/matrix-react-sdk/blob/d821323e5cb5f57231b91cdc7887c249ec522ec9/cypress/e2e/crypto/decryption-failure.spec.ts#L121-L126

The command above has Percy takes a snapshot of `.mx_DecryptionFailureBar` in `320px` and `640px`, in order to check whether buttons, messages, headlines, etc. would not overflow (they actually overflow right now per [this snapshot on 320px](https://percy.io/dfde73bd/matrix-react-sdk/builds/26217749/search/preview/1460385466?browser=firefox&searchParam=decry&viewLayout=overlay&viewMode=new&width=320), which should be fixed once https://github.com/matrix-org/matrix-react-sdk/pull/10360 is merged.)

Since `percySnapshotElement` is a custom command, I guess the information above should indeed be worth be noted on `docs/cypress.md`.